cli/cmd/doc.go

```diff
@@ -0,0 +1,91 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"sigs.k8s.io/yaml"
+)
+
+type cmdOption struct {
+	Name         string
+	Shorthand    string `yaml:",omitempty"`
+	DefaultValue string `yaml:"default_value,omitempty"`
+	Usage        string `yaml:",omitempty"`
+}
+
+type cmdDoc struct {
+	Name             string
+	Synopsis         string      `yaml:",omitempty"`
+	Description      string      `yaml:",omitempty"`
+	Options          []cmdOption `yaml:",omitempty"`
+	InheritedOptions []cmdOption `yaml:"inherited_options,omitempty"`
+	Example          string      `yaml:",omitempty"`
+	SeeAlso          []string    `yaml:"see_also,omitempty"`
+}
+
+var cmdList []cmdDoc
```

Comment:
In generally I'm anti-global variables, especially in this package, which includes all of our CLI subcommands. It seems like you could remove this one pretty easily by having `generateDocs` return the list that it populates. Something like:

```diff
diff --git a/cli/cmd/doc.go b/cli/cmd/doc.go
index b172b908..dc744339 100644
--- a/cli/cmd/doc.go
+++ b/cli/cmd/doc.go
@@ -28,8 +28,6 @@ type cmdDoc struct {
 	SeeAlso          []string    `yaml:"see_also,omitempty"`
 }
 
-var cmdList []cmdDoc
-
 func newCmdDoc() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:    "doc",
@@ -37,7 +35,8 @@ func newCmdDoc() *cobra.Command {
 		Short:  "Generate YAML documentation for the Linkerd CLI",
 		Args:   cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if err := generateDocs(RootCmd); err != nil {
+			cmdList, err := generateDocs(RootCmd)
+			if err != nil {
 				return err
 			}
 
@@ -57,35 +56,39 @@ func newCmdDoc() *cobra.Command {
 
 // generateDocs takes a command and recursively walks the tree of commands,
 // adding each as an item to cmdList.
-func generateDocs(cmd *cobra.Command) error {
+func generateDocs(cmd *cobra.Command) ([]cmdDoc, error) {
+	var cmdList []cmdDoc
+
 	for _, c := range cmd.Commands() {
 		if !c.IsAvailableCommand() || c.IsAdditionalHelpTopicCommand() {
 			continue
 		}
 
-		if err := generateDocs(c); err != nil {
-			return err
+		subList, err := generateDocs(c)
+		if err != nil {
+			return cmdList, err
 		}
+		cmdList = append(cmdList, subList...)
 	}
 
 	var buf bytes.Buffer
 
 	if err := doc.GenYaml(cmd, io.Writer(&buf)); err != nil {
-		return err
+		return cmdList, err
 	}
 
 	var doc cmdDoc
 	if err := yaml.Unmarshal(buf.Bytes(), &doc); err != nil {
-		return err
+		return cmdList, err
 	}
 
 	// Cobra names start with linkerd, strip that off for the docs.
-	doc.Name = strings.Replace(doc.Name, "linkerd ", "", 1)
+	doc.Name = strings.TrimPrefix(doc.Name, "linkerd ")
 
 	// Don't include the root command.
 	if doc.Name != "linkerd" {
 		cmdList = append(cmdList, doc)
 	}
 
-	return nil
+	return cmdList, nil
 }
```

Mind updating?